test/serializer/pure-functions/AbstractObject2.js

```diff
@@ -0,0 +1,19 @@
+let c = global.__abstract ? __abstract('boolean', '(true)') : true;
+let unknownObj = global.__abstract ? __abstract('object', '({})') : {};
+
+function test(fn) {
+  let knownObj = {x:1};
+  let conditionalUnknownObj = c ? knownObj : unknownObj;
+  fn(conditionalUnknownObj); // This must havoc the known obj
```

Comment:
I guess here the first half of visitAbstractValue, since we don't have an overapproximation of a concrete values domain. As far as I can read, this would also potentially havoc  objects referenced in `c`. So I think another refinement and test case is needed to avoid and test that.

Reply:
I could add a special case for conditionals like this but there is really a large number of similar operations. I can’t special case them all. I just thought of this one as a test case.